### ISSUE TITLE
when python version >3.10, precompile pattern will fail

### ISSUE THREAD

**Issue by mliu98** [BCHSI/philter-deidstable1_mirror#2]
Hi,

I just noticed  when python version is larger than 3.10, the precompile pattern function in philter.py will fail. 

For example, this is one of the given regex, and I found that most of the regexes in filters/regex/safe will fail:
`\b(?i)a1c\b`
if we run 
```python
re.compile(`\b(?i)a1c\b`)
``` 
with version <=3.10, it only gives deprecation warning, but with python version > 3.10, it will throw an exception "global flags not at the start of the expression at position 1".
![image](https://github.com/BCHSI/philter-deidstable1_mirror/assets/45385596/dd5d3626-a94c-4940-b92f-190045b43c57)

The following are the related snippets - philter.py->precompile():
```python
def precompile(self, filepath):
        """ precompiles our regex to speed up pattern matching"""
        print(filepath)
        # for line in open(filepath,"r"):
        resource_stream = pkg_resources.resource_stream(__name__, filepath)
        for line_bytes in resource_stream:
            line = line_bytes.decode("utf-8")
            if not line.strip().startswith("#"):
               regex = line.strip()
        re_compiled = None
        print(filepath, regex)

        with warnings.catch_warnings(): #NOTE: this is not thread safe! but we want to print a more detailed warning message
            warnings.simplefilter(action="error", category=FutureWarning) # in order to print a detailed message
            try:
                re_compiled = re.compile(regex)
            except FutureWarning as warn:
                print("FutureWarning: {0} in file ".format(warn) + filepath)
                warnings.simplefilter(action="ignore", category=FutureWarning)
                re_compiled = re.compile(regex) # assign nevertheless
        return re_compiled
``` 

So I would suggest either specifying python version in readme.md, or considering modifying regexes and the precompile function.

For example:
From
```python
re.compile(`\b(?i)a1c\b`)
```
To 
```python
re.compile(`\ba1c\b`, re.IGNORECASE)
```
will work in higher version of Python

Thanks,
Mengmeng

**Comment by RedChrists** [comment]
Thanks for reporting this issue. We are aiming to update the repo in a couple of weeks with updated regular expressions that conform with newer Python versions.

**Comment by RedChrists** [comment]
This should be fixed with the latest release v1.2024
